app/controllers/api/v1/applications_controller.rb

```diff
@@ -9,11 +9,16 @@ def index
       end
 
       def show
-        render json: to_json(applications_query.application(ecf_id: application_params[:ecf_id]))
+        render json: to_json(application)
       end
 
       def accept = head(:method_not_allowed)
-      def reject = head(:method_not_allowed)
+
+      def reject
+        service = Applications::Reject.new(application:)
+
+        render_from_service(service)
```

Comment:
If we want to wrap the `service.call` in a database transaction with some other stuff (say recording an event) would it make sense to do it like this - putting the render within the transaction feels a bit weird:

```suggestion
        Application.transaction do
          render_from_service(service)
          SomeEvent.create("foo")
        end
```

It might make more sense if the `call` was done within `#reject` instead:

```suggestion
        Application.transaction do
          service.call
          SomeEvent.create("foo")
        end
        
        render_from_service(service)
```

Reply:
I would say the transaction and events should be part of the service and `service.call` method, instead of it being in the controller.

Reply:
eg:
```ruby
    def call
      return self unless valid?
      Application.transaction do
          application.update!(lead_provider_approval_status: "rejected")
          SomeEvent.create("foo")
      end
      application
    end
```

Reply:
I prefer this too, makes it easier to test as well to avoid those being in the request spec

Reply:
> I would say the transaction and events should be part of the service and service.call method, instead of it being in the controller.

Yeah you're right, I misread 🤦🏽‍♂️